### ISSUE TITLE
Set up linting

### ISSUE THREAD

**Issue by jasonaowen** [PhilanthropyDataCommons/auth#4]
We should set up CI to run unit tests and a linter.

Part of this is deciding on a style guide for our linter. It's been a few years, but I think the most recent CheckStyle configuration I used was on the PSM: https://github.com/SolutionGuidance/psm/blob/master/psm-app/checkstyle.xml . I think my opinions on Java style are less strongly held now than they were then, though, so maybe we should just use the Google configuration?

This Stack Overflow answer shows [how to configure CheckStyle with Google checks in Gradle](https://stackoverflow.com/a/67513272).

**Comment by bickelj** [comment]
Things still needed for a CI/CD pipeline for `twilio-keycloak-provider`
- [x] Versioning binaries
- [x] Publishing binaries (e.g. to GHCR)
- [X] Reorganize the directories in `twilio-keycloak-provider`
- [x] Linting (this ticket, #4)
- [ ] Easier viewing of code coverage results
- [x] Use gradle-build-action for faster builds

At a higher level:
- [X] Use gradle at the top level of this repo, let each subdirectory produce a jar, create a simple build script for a `theme` jar (or other appropriate name) in the subdirectory for the theme.

**Comment by jasonaowen** [comment]
That's a comprehensive list; thanks, @bickelj!

I was imagining that this issue would only be for tracking work on CI, and _not_ on CD - that is, running tests and linting, and not versioning and publishing! In case that helps this feel less overwhelming. :smiley:

**Comment by bickelj** [comment]
@jasonaowen Suppose we repurpose/rename this one to adding linting since the test run workflows (admittedly perhaps inappropriately) pushed with #3 and create the others by name as well?

**Comment by jasonaowen** [comment]
Yes, I had imagined linting as being a part of this, but I should have more explicitly said so!

**Comment by bickelj** [comment]
Done in #142, other checkboxes ticked in #132, #133, #141.
